framework/tests/src/tests/defaulting/FieldQual.java

```diff
@@ -0,0 +1,48 @@
+package tests.defaulting;
+
+import com.sun.source.tree.Tree.Kind;
+import org.checkerframework.framework.qual.*;
+
+import java.lang.annotation.*;
+
+public class FieldQual {
+
+    @DefaultQualifierInHierarchy
+    @Documented
+    @Retention(RetentionPolicy.RUNTIME)
+    @SubtypeOf({})
+    @Target({ElementType.TYPE_USE})
+    public static @interface F_TOP {
+    }
+
+
+    @Documented
+    @Retention(RetentionPolicy.RUNTIME)
+    @SubtypeOf(F_TOP.class)
+    @ImplicitFor(trees = Kind.MEMBER_SELECT)
+    @Target({ElementType.TYPE_USE})
+    /**
+     * This annotation should not be applied implicitly
+     * because the type of a member select should
+     * the type of the member.
```

Comment:
I don't understand the point of this whole test.
Should the `@ImplicitFor` be forbidden? This comment seems to say that it should never actually apply, which is quite misleading behavior. There are many other Tree.Kind that do not make sense for `@ImplicitFor` and default qualifier tree specifications.
Is the point of this pull request that ImplicitFor with MEMBER_SELECT used to do what one would expect (use the qualifier for member select trees) and change it to silently ignore such implicits?
How does this help inference?